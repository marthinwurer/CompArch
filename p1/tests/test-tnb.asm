; test the tnb instruction
	.entry	start
;
	.org	0x20
zero:	.word	0
one:	.word	0x00001
five:	.word	0x00005
neg1:	.word	0xfffff
neg5:	.word	0xffffb
error:	.word	0x00bad
;
	.org	0x100
start:
	ldb	zero
	tnb	err1
	tr	test2
err1:	ldb	error
;
test2:	ldb	one
	tnb	err2
	tr	test3
err2:	ldb	error
;
test3:	ldb	five
	tnb	err3
	tr	test4
err3:	ldb	error
;
test4:	ldb	neg1
	tnb	test5
err4:	ldb	error
;
test5:	ldb	zero
	tnb	err5
	tr	test6
err5:	ldb	error
;
test6:	ldb	neg5
	tnb	out
err6:	ldb	error
;
out:
	halt
