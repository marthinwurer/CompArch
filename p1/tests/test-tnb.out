CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 100
100:  11020  LDB   020  A[00000]  B[00000]  XR[000]
101:  35103  TNB   103  A[00000]  B[00000]  XR[000]
102:  30104  TR    104  A[00000]  B[00000]  XR[000]
104:  11021  LDB   021  A[00000]  B[00000]  XR[000]
105:  35107  TNB   107  A[00000]  B[00001]  XR[000]
106:  30108  TR    108  A[00000]  B[00001]  XR[000]
108:  11022  LDB   022  A[00000]  B[00001]  XR[000]
109:  3510b  TNB   10b  A[00000]  B[00005]  XR[000]
10a:  3010c  TR    10c  A[00000]  B[00005]  XR[000]
10c:  11023  LDB   023  A[00000]  B[00005]  XR[000]
10d:  3510f  TNB   10f  A[00000]  B[fffff]  XR[000]
10f:  11020  LDB   020  A[00000]  B[fffff]  XR[000]
110:  35112  TNB   112  A[00000]  B[00000]  XR[000]
111:  30113  TR    113  A[00000]  B[00000]  XR[000]
113:  11024  LDB   024  A[00000]  B[00000]  XR[000]
114:  35116  TNB   116  A[00000]  B[ffffb]  XR[000]
116:  00000  HALT       A[00000]  B[ffffb]  XR[000]

Machine Halted - HALT instruction executed

Simulated time 90 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
